packages/react-app/src/components/HardhatEVM.jsx

```diff
@@ -0,0 +1,175 @@
+import { SendOutlined } from "@ant-design/icons";
+import { Button, Input, Tooltip, Drawer, Form, Row, Col, Select, DatePicker, Divider } from "antd";
+import { useLookupAddress } from "eth-hooks";
+import React, { useCallback, useEffect, useState } from "react";
+import Blockies from "react-blockies";
+import { Transactor } from "../helpers";
+import Wallet from "./Wallet";
+import {
+  useOnBlock,
+} from "../hooks";
+import { formatEther, parseEther } from "@ethersproject/units";
+import { parse } from "graphql";
+
+const { Option } = Select;
+const { utils, BigNumber } = require("ethers");
+
+/*
+  ~ What it does? ~
+
+  
+
+  ~ How can I use? ~
+
+  <HardhatEVM
+    localProvider={localProvider}
+  />
+
+  ~ Features ~
+*/
+
+export default function HardhatEVM(props) {
+  const [address, setAddress] = useState();
+  const [visible, setVisible] = useState(false);
+  const [chainTimestamp, setChainTimestamp] = useState();
+  const [accountBalance, setAccountBalance] = useState();
+
+  const localProvider = props.localProvider;
+
+  useOnBlock(localProvider, () => {
+    updateTime();
+  });
+
+  async function updateTime(){
+    const time = await localProvider.getBlock();
+    setChainTimestamp(new Date(time.timestamp * 1000).toLocaleString("en-US"));
+  }
+
+  function onClose(){
+    setVisible(false);
+  }
+
+  async function showDrawer(){  
+    setField('hardhat_setBalance_address', props.address);
+    const newBalance = await localProvider.getBalance(props.address);
+    setAccountBalance(formatEther(newBalance));
+    setVisible(true);
+  }
+
+  async function increaseTime(){
+    await localProvider.send("evm_increaseTime", [parseInt(form["increaseTimeSeconds"])]);
+    await localProvider.send("evm_mine");
+    updateTime();
+  }
+
+   async function setNextBlockTimestamp(){
+    await localProvider.send("evm_setNextBlockTimestamp", [parseInt(form["nextBlockTimestamp"])]);
+    await localProvider.send("evm_mine");
+    updateTime();
+  }
+
+  async function hardhatSetBalance(){
+    const aNum = parseInt(parseEther(form["hardhat_setBalance_amount"]));
+    console.log(aNum.toString(16));
+    await localProvider.send("hardhat_setBalance", [form["hardhat_setBalance_address"], "0x"+aNum.toString(16)]);
+    await localProvider.send("evm_mine");
+    const newBalance = await localProvider.getBalance(form["hardhat_setBalance_address"]);
+    setAccountBalance(formatEther(newBalance));
+  }
+
+  const [ form, setForm ] = useState({})
+  const setField = (field, value) => {
```

Comment:
the recommended react method is the use callback of setState function, rather than using the useState value

```
setForm(oldValue => { ...oldValue, [field]:value
});
```